test_conformance/api/test_kernels.cpp

```diff
@@ -704,3 +722,65 @@ REGISTER_TEST(negative_set_immutable_memory_to_writeable_kernel_arg)
 
     return TEST_PASS;
 }
+
+REGISTER_TEST(negative_set_read_write_image_arg)
+{
+    cl_int error = CL_SUCCESS;
+    clProgramWrapper program;
+    clKernelWrapper kernels[2];
+    clMemWrapper images[2], buffer;
+    const char *test_kernels[2] = { sample_read_only_image_test_kernel,
+                                    sample_write_only_image_test_kernel };
+    constexpr cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+    const int size_dim = 128;
+
+    // Setup the test
+    error = create_single_kernel_helper(context, &program, nullptr, 2,
+                                        test_kernels, nullptr);
+    test_error(error, "Unable to build test program");
+
+    kernels[0] = clCreateKernel(program, "read_only_image_test", &error);
+    test_error(error,
+               "Unable to get read_only_image_test kernel for built program");
+
+    kernels[1] = clCreateKernel(program, "write_only_image_test", &error);
+    test_error(error,
+               "Unable to get write_only_image_test kernel for built program");
+
+    buffer = clCreateBuffer(context, CL_MEM_WRITE_ONLY, sizeof(cl_uint) * 4,
+                            nullptr, &error);
+    test_error(error, "clCreateBuffer failed");
+
+    images[0] = create_image_2d(context, CL_MEM_READ_ONLY, &formats, size_dim,
+                                size_dim, 0, nullptr, &error);
+    test_error(error, "create_image_2d failed");
+
+    images[1] = create_image_2d(context, CL_MEM_WRITE_ONLY, &formats, size_dim,
+                                size_dim, 0, nullptr, &error);
+    test_error(error, "create_image_2d failed");
+
+    error = clSetKernelArg(kernels[0], 1, sizeof(buffer), &buffer);
+    test_error(error, "clSetKernelArg failed");
+
+    // Run the test
+    error = clSetKernelArg(kernels[0], 0, sizeof(images[0]), &images[0]);
+    test_failure_error_ret(error, CL_INVALID_ARG_VALUE,
```

Comment:
I'm not going to lose a ton of sleep over this, but it seems odd to me that we need to use `test_failure_error_ret` here vs. `test_failure_error`.